xio-core/src/main/java/com/xjeffrose/xio/application/ApplicationConfig.java

```diff
@@ -104,11 +106,19 @@ public ServerChannelConfiguration serverChannelConfig() {
         bossThreads, bossNameFormat, workerThreads, workerNameFormat);
   }
 
+  // TODO(CK): refactor this into a ZooKeeperConfig that the ZkClient constructor will accept
   public ZkClient zookeeperClient() {
     if (zookeeperCluster.isEmpty()) {
       return new NullZkClient();
     } else {
-      return new ZkClient(zookeeperCluster);
+      if (zookeeperCluster.startsWith("exhibitor:")) {
+        String[] values = zookeeperCluster.replace("exhibitor:", "").split(":");
+        int restPort = Integer.parseInt(values[0]);
```

Comment:
maybe bounds check array before indexing into it

Reply:
I'm fine with this throwing an exception if you did something silly. We can clean this up in a ZooKeeperConfig class later.